builtin/providers/google/resource_bigquery_dataset.go

```diff
@@ -0,0 +1,87 @@
+package google
+
+import (
+	"google.golang.org/api/bigquery/v2"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceBigQueryDataset() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceBigQueryDatasetCreate,
+		Read:   resourceBigQueryDatasetRead,
+		Update: resourceBigQueryDatasetUpdate,
+		Delete: resourceBigQueryDatasetDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
```

Comment:
For starters
* `datasetReference` is described as being required. This resource as is shouldn't work as is.

Otherwise,
* `access[]`
* `friendlyName` (which you've called `name`)
* `description`
* `defaultTableExpirationMs`
* `location`
are optional, but nice to have.

When it comes to exported attributes
* `self_link` is important to include as `computed`, since other resources often refer to one another by GCE assigned URI.
* `id` for similar reasons.
* `lastModifiedTime` can be useful for people examining infrastructure managed by Terraform.